packages/EasyCore/tests/Doctrine/Dispatchers/DeferredEntityEventDispatcherTest.php

```diff
@@ -266,6 +268,59 @@ public function testDispatchSucceedsWithEntities(): void
         self::assertSame([], $this->getPrivatePropertyValue($deferredEntityEventDispatcher, 'entityUpdates'));
     }
 
+    public function testDispatchSucceedsWithOneEventForNewEntity(): void
+    {
+        /** @var object $newEntity */
+        $newEntity = $this->prophesize(stdClass::class)->reveal();
+        $oid = \spl_object_hash($newEntity);
+        $entityCreatedEvent = new EntityCreatedEvent($newEntity);
+        $entityUpdatedEvent = new EntityUpdatedEvent($newEntity);
+        $eventDispatcher = $this->prophesize(EventDispatcherInterface::class);
+        $eventDispatcher->dispatch($entityCreatedEvent)
+            ->willReturnArgument(0);
+        $eventDispatcher->dispatch($entityUpdatedEvent)
+            ->willReturnArgument(0);
+        /** @var \Psr\EventDispatcher\EventDispatcherInterface $eventDispatcherReveal */
+        $eventDispatcherReveal = $eventDispatcher->reveal();
+        $deferredEntityEventDispatcher = new DeferredEntityEventDispatcher($eventDispatcherReveal);
+
+        $deferredEntityEventDispatcher->deferInsertions([$oid => $newEntity], 0);
+        $deferredEntityEventDispatcher->deferUpdates([$oid => $newEntity], 0);
+        $deferredEntityEventDispatcher->deferUpdates([$oid => $newEntity], 1);
+        $deferredEntityEventDispatcher->dispatch();
+
+        $eventDispatcher->dispatch($entityCreatedEvent)
+            ->shouldHaveBeenCalledOnce();
+        $eventDispatcher->dispatch($entityUpdatedEvent)
+            ->shouldNotHaveBeenCalled();
+        self::assertSame([], $this->getPrivatePropertyValue($deferredEntityEventDispatcher, 'entityInsertions'));
+        self::assertSame([], $this->getPrivatePropertyValue($deferredEntityEventDispatcher, 'entityUpdates'));
+    }
+
+    public function testDispatchSucceedsWithOneEventForUpdatedEntity(): void
+    {
+        /** @var object $existedEntity */
+        $existedEntity = $this->prophesize(stdClass::class)->reveal();
+        $oid = \spl_object_hash($existedEntity);
+        $entityUpdatedEvent = new EntityUpdatedEvent($existedEntity);
+        $eventDispatcher = $this->prophesize(EventDispatcherInterface::class);
+        $eventDispatcher->dispatch($entityUpdatedEvent)
+            ->willReturnArgument(0);
+        /** @var \Psr\EventDispatcher\EventDispatcherInterface $eventDispatcherReveal */
+        $eventDispatcherReveal = $eventDispatcher->reveal();
```

Comment:
We don't need a separate variable for this value. Please get rid of this variable here and in other tests.

Reply:
Currently, we need it to avoid errors from phpstran:

```
 -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  packages/EasyCore/tests/Doctrine/Dispatchers/DeferredEntityEventDispatcherTest.php:255
 -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  - '#Parameter \#1 \$eventDispatcher of class EonX\\EasyCore\\Doctrine\\Dispatchers\\DeferredEntityEventDispatcher constructor expects Psr\\EventDispatcher\\EventDispatcherInterface, object given#'
 -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

Can we fix it in a separate PR for all tests?